src/rdkafka_admin.c

```diff
@@ -9009,3 +9010,284 @@ void rd_kafka_DescribeCluster(rd_kafka_t *rk,
+                                         elect_leaders->partitions);
+}
+
+void rd_kafka_ElectLeaders_destroy(rd_kafka_ElectLeaders_t *elect_leaders) {
+        rd_kafka_topic_partition_list_destroy(elect_leaders->partitions);
+        rd_free(elect_leaders);
+}
+
+static void rd_kafka_ElectLeaders_free(void *ptr) {
+        rd_kafka_ElectLeaders_destroy(ptr);
+}
+
+/**
+ * @brief Creates a new elect leaders result object with the given
+ *        \p error.
+ */
+static rd_kafka_ElectLeadersResult_t *
+rd_kafka_ElectLeadersResult_new(rd_kafka_resp_err_t err,
+                                rd_list_t *partitions) {
+
+        rd_kafka_ElectLeadersResult_t *result;
+        result = rd_calloc(1, sizeof(*result));
+
+        result->err = err;
+        rd_list_init_copy(&result->partitions, partitions);
+        rd_list_copy_to(&result->partitions, partitions,
+                        rd_kafka_topic_partition_result_copy_opaque, NULL);
+        return result;
+}
+
+const rd_kafka_ElectLeadersResult_t *
+rd_kafka_ElectLeaders_result(const rd_kafka_ElectLeaders_result_t *result) {
+        return (const rd_kafka_ElectLeadersResult_t *)rd_list_elem(
+            &result->rko_u.admin_result.results, 0);
+}
+
+rd_kafka_resp_err_t
+rd_kafka_ElectLeadersResult_error(const rd_kafka_ElectLeadersResult_t *result) {
+        return result->err;
+}
+
+const rd_kafka_topic_partition_result_t **
+rd_kafka_ElectLeadersResult_partitions(
+    const rd_kafka_ElectLeadersResult_t *result,
+    size_t *cntp) {
+        *cntp = rd_list_cnt(&result->partitions);
+        return (const rd_kafka_topic_partition_result_t **)
+            result->partitions.rl_elems;
+}
+
+void rd_kafka_ElectLeadersResult_destroy(
+    rd_kafka_ElectLeadersResult_t *result) {
+        rd_list_destroy(&result->partitions);
+        rd_free(result);
+}
+
+static void rd_kafka_ElectLeadersResult_free(void *ptr) {
+        rd_kafka_ElectLeadersResult_destroy(ptr);
+}
+
+/**
+ * @brief Parse ElectLeadersResponse and create ADMIN_RESULT op.
+ */
+static rd_kafka_resp_err_t
+rd_kafka_ElectLeadersResponse_parse(rd_kafka_op_t *rko_req,
+                                    rd_kafka_op_t **rko_resultp,
+                                    rd_kafka_buf_t *reply,
+                                    char *errstr,
+                                    size_t errstr_size) {
+        const int log_decode_errors           = LOG_ERR;
+        rd_kafka_op_t *rko_result             = NULL;
+        rd_kafka_ElectLeadersResult_t *result = NULL;
+        int16_t top_level_error_code          = 0;
+        int32_t TopicArrayCnt;
+        rd_list_t partitions_arr;
+        rd_kafka_ElectLeaders_t *elect_leaders_request =
+            rko_req->rko_u.admin_request.args.rl_elems[0];
+        rd_kafka_topic_partition_list_t *topic_partitions =
+            rd_kafka_topic_partition_list_copy(
+                elect_leaders_request->partitions);
+        int i;
+        int idx;
```

Comment:
Change this to `j` again as we are not using `idx` for destroying partitions_arr anymore.